File: uebung01/BankServer/BankServer/Account.cs
using System;
using System.Globalization;

namespace BankServer
{
    public class Account
    {

        private static int _sequence;

        private readonly string _owner;

        public string Owner
        {
            get { return _owner; }
        }

        private readonly string _number;

        public string Number
        {
            get { return _number;  }
        }

        public double Balance { get; private set; }

        public bool IsActive { get; set; }

		public Account(String owner) {
		    Balance = 0;
		    _owner = owner;
		    _number = _sequence++.ToString(CultureInfo.InvariantCulture);
		    IsActive = true;
		}


		public void Deposit(double amount) {
            if (!IsActive)
            {
				throw new InactiveException();
			}
			
			if (amount < 0) {
				throw new ArgumentException();
			}
			
			Balance += amount;
		}

		public void Withdraw(double amount) {
			if (!IsActive) {
				throw new InactiveException();
			}
			
			if (amount < 0) {
				throw new ArgumentException("");
			}
			
			if (amount > Balance) {
				throw new OverdrawException();
			}
			
			Balance -= amount;
		}

	}
}

File: uebung02/BankServlet/src/bank/implementation/Account.java
package bank.implementation;

import java.util.UUID;

import bank.InactiveException;
import bank.OverdrawException;

public final class Account implements bank.Account {
	private String number;
	private String owner;
	private double balance;
	private boolean active = true;

	Account(final String owner) {
		this.owner = owner;
		this.number = UUID.randomUUID().toString();
	}

	@Override
	public double getBalance() {
		return balance;
	}

	@Override
	public String getOwner() {
		return owner;
	}

	@Override
	public String getNumber() {
		return number;
	}

	@Override
	public boolean isActive() {
		return active;
	}
	
	void setActive(final boolean active) {
		this.active = active;
	}


	@Override
	public void deposit(final double amount) 
			throws InactiveException {
		if (!this.isActive()) {
			throw new InactiveException("Account is closed!");
		}
		
		if (amount < 0) {
			throw new IllegalArgumentException("Amount can't be negative!");
		}
		
		this.balance += amount;
	}

	@Override
	public void withdraw(final double amount) 
			throws InactiveException, OverdrawException {
		if (!this.isActive()) {
			throw new InactiveException("Account is closed!");
		}
		
		if (amount < 0) {
			throw new IllegalArgumentException("Amount can't be negative!");
		}
		
		if (amount > balance) {
			throw new OverdrawException("Not enough money on account!");
		}
		
		this.balance -= amount;
	}

}


